core/src/main/java/io/questdb/cairo/sql/VirtualRecord.java

```diff
@@ -208,6 +208,16 @@ public long getUpdateRowId() {
         return base.getUpdateRowId();
     }
 
+    @Override
+    public long getUuidHi(int col) {
```

Comment:
when i say inefficient API, i mean we need to have two. One for when contiguous memory is available and we can do less to access values and another this one.